pynamodb/models.py

```diff
@@ -267,6 +264,18 @@ def __init__(self, name, bases, namespace, discriminator=None) -> None:
                 }
                 cls.DoesNotExist = type('DoesNotExist', (DoesNotExist, ), exception_attrs)
 
+    @staticmethod
+    def _initialize_indexes(cls):
+        """
+        Initialize indexes on the class.
+        """
+        cls._indexes = {}
+        for name, index in getmembers(cls, lambda o: isinstance(o, Index)):
+            index.Meta.model = cls
+            if not hasattr(index.Meta, "index_name"):
+                index.Meta.index_name = name
+            cls._indexes[index.Meta.index_name] = index
```

Comment:
This keeps the current broken behavior where you can have index name collisions.